Updated API Platform docs. (#3691)
spryker/spryker-docs: docs/dg/dev/architecture/api-platform/code-buckets.md

Resource schemas follow the pattern: `{resource-name}.resource.yml`
Validation schemas follow the pattern: `{resource-name}.validation.yml`

> Each YAML file contains exactly one resource definition. To create a CodeBucket variant, place its schema in a **separate module directory** named after the variant (for example, `StoreEU`, `StoreAT`) and set the `codeBucket:` property inside the file. The filename stays the same across all variants — only the parent module directory and the `codeBucket:` value differ.

Reviewer: ```suggestion
Each YAML file contains exactly one resource definition. To create a CodeBucket variant, place its schema in a **separate module directory** named after the variant (for example, `StoresApiEU`, `StoresApiAT`) and set the `codeBucket:` property inside the file. The filename stays the same across all variants — only the parent module directory and the `codeBucket:` value differ.
```

```MARKDOWN

> src/Pyz/Glue/Store/resources/api/backend/           # Base variant module

Reviewer: ```suggestion
src/Pyz/Glue/StoresApi/resources/api/backend/           # Base variant module
```

├── stores.resource.yml                              # No codeBucket property
└── stores.validation.yml

> src/Pyz/Glue/StoreEU/resources/api/backend/         # EU variant module

Reviewer: ```suggestion
src/Pyz/Glue/StoresApiEU/resources/api/backend/         # EU variant module
```

├── stores.resource.yml                              # codeBucket: EU
└── stores.validation.yml                            # codeBucket: EU

> src/Pyz/Glue/StoreAT/resources/api/backend/         # AT variant module

Reviewer: ```suggestion
src/Pyz/Glue/StoresApiAT/resources/api/backend/         # AT variant module
```

      description: "Store timezone"
```

**Step 2: Define the CodeBucket variant in a separate module**

Create a new module directory whose name carries the CodeBucket suffix (`StoreEU`) and place the variant's schema there. The filename stays `stores.resource.yml`; the `codeBucket:` property inside the file declares which variant it belongs to.

> `src/Pyz/Glue/StoreEU/resources/api/backend/stores.resource.yml`

Reviewer: ```suggestion
`src/Pyz/Glue/StoresApiEU/resources/api/backend/stores.resource.yml`
```

      description: "VAT registration number"
```

{% info_block infoBox "One resource per file" %}

Each schema file must contain exactly one top-level `resource:` block. CodeBucket variants live in their own files in their own module directories — they cannot be stacked inside a single YAML document.

{% endinfo_block %}

**Step 3: Create the variant's validation schema**

Place the matching validation file alongside the variant resource, in the same module directory. Set `codeBucket:` at the root so the parser pairs it with the EU resource variant.

> `src/Pyz/Glue/StoreEU/resources/api/backend/stores.validation.yml`

Reviewer: ```suggestion
`src/Pyz/Glue/StoresApiEU/resources/api/backend/stores.validation.yml`
```

```yaml
# ✅ Good - Base is complete

> # src/Pyz/Glue/Store/resources/api/backend/stores.resource.yml

Reviewer: ```suggestion
# src/Pyz/Glue/StoresApi/resources/api/backend/stores.resource.yml
```

      type: string
    timezone:
      type: string
```

```yaml
# ✅ EU variant lives in its own module directory

> # src/Pyz/Glue/StoreEU/resources/api/backend/stores.resource.yml

Reviewer: ```suggestion
# src/Pyz/Glue/StoresApiEU/resources/api/backend/stores.resource.yml
```
